test_runner/regress/test_metric_collection.py

```diff
@@ -141,20 +107,50 @@ def get_num_remote_ops(file_kind: str, op_kind: str) -> int:
             total += sample[2]
         return int(total)
 
+    remote_uploaded = 0
+
     # upload some data to remote storage
     if remote_storage_kind == RemoteStorageKind.LOCAL_FS:
         wait_for_last_flush_lsn(env, endpoint, tenant_id, timeline_id)
         pageserver_http = env.pageserver.http_client()
         pageserver_http.timeline_checkpoint(tenant_id, timeline_id)
         pageserver_http.timeline_gc(tenant_id, timeline_id, 10000)
-        global remote_uploaded
         remote_uploaded = get_num_remote_ops("index", "upload")
         assert remote_uploaded > 0
 
-    # wait longer than collecting interval and check that all requests are served
-    time.sleep(3)
+    # wait for the uploads to start happening
+    log.info("waiting for queue")
+    events = uploads.get()
     httpserver.check()
-    global metric_kinds_checked, checks
+    httpserver.stop()
```

Comment:
Expecting first ones to fail because there is no remote upload yet and so on.

I'll probably need to do "clear" and wait one more here, to order up correctly. Then the while loop below is no longer needed.